be/src/exec/pipeline/fragment_executor.cpp

```diff
@@ -141,23 +140,23 @@ Status FragmentExecutor::_prepare_workgroup(const TExecPlanFragmentParams& reque
             wg = WorkGroupManager::instance()->get_default_workgroup();
         }
         DCHECK(wg != nullptr);
-        if (_query_ctx->init_query(wg.get())) {
-            _wg = wg;
-        }
+        RETURN_IF_ERROR(_query_ctx->init_query(wg.get()));
+        _wg = wg;
```

Comment:
if a query_ctx created success and it has two fragment_ctx.
In this case if fragment 1 and fragment 2 both failed, _wg->decr_num_queries() will execute twice.

Reply:
The `_wg->decr_num_queries()`will only be executed after `QueryContextManager::remove`